Actions/Deploy/Deploy.ps1

```
@@ -100,6 +101,58 @@ function InstallOrUpgradeApps {
     }
 }
 
+function InstallUnknownDependencies {
+    Param(
+        [hashtable] $bcAuthContext,
+        [string] $environment,
+        [string[]] $apps,
+        [string] $installMode
+    )
+
+    Write-Host "Installing unknown dependencies: $($apps -join ', ')"
+    try {
+        $installedApps = Get-BcInstalledExtensions -bcAuthContext $bcAuthContext -environment $environment | Where-Object { $_.isInstalled }
+        # Run through all apps and install or upgrade AppSource apps first (and collect PTEs)
+        foreach($app in $apps) {
+            #The output of Sort-AppFilesByDependencies is in the format of "AppId:AppName"
+            $appId, $appName = $app.Split(':')
+            $appVersion = ""
+            if ($appName -like "*Tests-TestLibraries*") {
+                Write-Host "::WARNING::Tests-TestLibraries can't be installed"
+                continue
+            }
+            if ($appName -match "_(\d+\.\d+\.\d+\.\d+)\.app$") {
+                $appVersion = $matches.1
+            } else {
+                throw "Version not found for unknown dependency $app"
```

when would we ever get in this situation?
and what should the user/partner do about that?
wouldn't it be better to just silently ignore if there is a dependency we cannot find/use - then the installation will fail later with a more explanatory error message: Dependency xxx is missing.

---

I agree it makes sense not to throw an error, but it might still be helpful to write $appName as either a warning or just regular message in the terminal?
If it does actually fail to find the version, it might be helpful for us for debugging if the partner can show us the app name incase something actually goes wrong.

But you know the system better than me, so if you are confident that the app name returned by Sort-AppFilesByDependencies always include the version in correct format, I am happy to remove the else part completely.

---

You are right that in debugging scenarios, it would be good to have a note specifying that this dependency from that app doesn't include the version number, ignoring dependency, so basically a Write-Host instead of the throw and then continue.